deployment/ansible/roles/ckan-odp-configuration/templates/nginx_ckan.j2

```diff
@@ -0,0 +1,25 @@
+proxy_cache_path /tmp/nginx_cache levels=1:2 keys_zone=cache:30m max_size=250m;
+proxy_temp_path /tmp/nginx_proxy 1 2;
+
+server {
+    listen 80;
+    return 301 https://$host$request_uri;
+}
+
+server {
+    listen 443 ssl;
+    ssl_certificate {{ ckan_ssl_cert | default('/etc/ssl/certs/ssl-cert-snakeoil.pem') }};
+    ssl_certificate_key {{ ckan_ssl_key | default('/etc/ssl/private/ssl-cert-snakeoil.key') }};
+    client_max_body_size 100M;
+    location / {
+        proxy_pass http://127.0.0.1:8080/;
+        proxy_set_header Host $host;
+        proxy_cache cache;
+        proxy_cache_bypass $cookie_auth_tkt;
+        proxy_no_cache $cookie_auth_tkt;
+        proxy_cache_valid 30m;
+        proxy_cache_key $host$scheme$proxy_host$request_uri;
+        # In emergency comment out line to force caching
+        # proxy_ignore_headers X-Accel-Expires Expires Cache-Control;
+    }
+}
```

Comment:
If this file is just copied from the default config that the CKAN installer sets up, it might not be a bad idea to template it no matter what, and then wrap the SSL-specific parts of the template in guards so that they're only outputted if the `ckan_ssl` flag is set. That way it's already there if we need to make other nginx changes.